File: CS/Basic/recursion/SimpleSummation.go
package main

// import (
// 	"fmt"
// )

func simpleSummation(n int32) int32 {
	if n == 0 {
		return 0
	}
	return simpleSummation(n-1) + n
}

// func main() {
// 	var n int32
// 	fmt.Scan(&n)
// 	fmt.Println(simpleSummation(n))
// }

File: CS/Basic/recursion/SimpleSummation.java
package CS.Basic.recursion;

public class SimpleSummation {
    public static int simpleSummation(int n) {
        if (n == 0) {
            return 0;
        }

        return simpleSummation(n - 1) + n;
    }

    public static void main(String[] args) {
        System.out.println(simpleSummation(100));
    }
}
